contrib/src/main/java/com/datatorrent/contrib/cassandra/AbstractUpsertOutputOperator.java

```diff
@@ -0,0 +1,1401 @@
+      LOG.info("Registering query support for " + currentQueryStr);
+      PreparedStatement preparedStatementForThisQuery = session.prepare(currentQueryStr);
+      preparedStatementTypes.put(currentIndexPos, preparedStatementForThisQuery);
+    }
+  }
+
+  private Map<Long, String> generatePreparedStatementsQueryStrings()
+  {
+    Map<Long, String> queryStrings = new HashMap<>();
+    //UPDATE keyspace_name.table_name USING option AND option SET assignment, assignment, ... WHERE row_specification
+    StringBuilder updateQueryRoot = new StringBuilder(" UPDATE " + connectionStateManager.getKeyspaceName() +
+        "." + connectionStateManager.getTableName() + " ");
+    String ttlSetString = " USING ttl :" + TTL_PARAM_NAME + " SET ";
+    // TTL set , Collections Append , List prepend
+    StringBuilder queryExpTTLSetCollAppendListPrepend = new StringBuilder(updateQueryRoot.toString());
+    queryExpTTLSetCollAppendListPrepend.append(ttlSetString);
+    buildNonPKColumnsExpression(queryExpTTLSetCollAppendListPrepend,
+        UpsertExecutionContext.ListPlacementStyle.PREPEND_TO_EXISTING_LIST,
+        UpsertExecutionContext.CollectionMutationStyle.ADD_TO_EXISTING_COLLECTION);
+    queryStrings.put(getSlotIndexForMutationContextPreparedStatement(EnumSet.of(
+        OperationContext.TTL_SET, OperationContext.COLLECTIONS_APPEND, OperationContext.LIST_PREPEND
+    )), queryExpTTLSetCollAppendListPrepend.toString());
+
+    // TTL set , Collections Append , List append
+    StringBuilder queryExpTTLSetCollAppendListAppend = new StringBuilder(updateQueryRoot.toString());
+    queryExpTTLSetCollAppendListAppend.append(ttlSetString);
+    buildNonPKColumnsExpression(queryExpTTLSetCollAppendListAppend,
+        UpsertExecutionContext.ListPlacementStyle.APPEND_TO_EXISTING_LIST,
+        UpsertExecutionContext.CollectionMutationStyle.ADD_TO_EXISTING_COLLECTION);
+    queryStrings.put(getSlotIndexForMutationContextPreparedStatement(EnumSet.of(
+        OperationContext.TTL_SET, OperationContext.COLLECTIONS_APPEND, OperationContext.LIST_APPEND
+    )), queryExpTTLSetCollAppendListAppend.toString());
+
+    // TTL set , Collections Remove
+    StringBuilder queryExpTTLSetCollRemove = new StringBuilder(updateQueryRoot.toString());
+    queryExpTTLSetCollRemove.append(ttlSetString);
+    buildNonPKColumnsExpression(queryExpTTLSetCollRemove,
+        UpsertExecutionContext.ListPlacementStyle.APPEND_TO_EXISTING_LIST, // Just in case user sets it
+        UpsertExecutionContext.CollectionMutationStyle.REMOVE_FROM_EXISTING_COLLECTION);
+    queryStrings.put(getSlotIndexForMutationContextPreparedStatement(EnumSet.of(
+        OperationContext.TTL_SET, OperationContext.COLLECTIONS_REMOVE, OperationContext.LIST_APPEND
+    )), queryExpTTLSetCollRemove.toString());
+
+    // TTL Not set , Collections Append , List prepend
+    StringBuilder queryExpTTLNotSetCollAppendListPrepend = new StringBuilder(updateQueryRoot.toString());
```

Comment:
will do. I too felt that the method is too "long" initially. Although splitting it meant passing in a lot of context for each split, it is perhaps not a bad idea  as it is only invoked once )